### DISCUSSION TITLE
How do I replace replace flake8 and isort with ruff using my null-ls config

### DISCUSSION THREAD

**Discussion by PyDataBlog** [NvChad/NvChad#2476] [Q&A]
am currently struggling to update my working python linter ruff in my NvChad null-ls sources. Here's my current implementation which is not working as I can't see any Python code sorted. I'm using the example config with this null-ls.confg in `lua>custom>configs>null-ls.lua`:

```lua
local null_ls = require "null-ls"

local b = null_ls.builtins

local ruff_args = {
  "--select",
  "B,C,E,F,W,T4,B9,I001",
  "--ignore",
  "E203,E266,E501,W503,F403,F401,E712,C901,E402,F405",
  "--max-line-length",
  "120",
  "--per-file-ignores",
  "__init__.py:E402 **/{tests,docs,tools}/*:E402",
  "--isort",
  "--isort-known-third-party",
  "",
  "--isort-known-first-party",
  "",
  "--isort-sections",
  "FUTURE,STDLIB,THIRDPARTY,FIRSTPARTY,LOCALFOLDER",
}

local sources = {

  --python
  b.formatting.black,
  b.diagnostics.ruff.with { extra_args = ruff_args },
}

null_ls.setup {
  debug = true,
  sources = sources,
}
```

**Comment by siduck** [comment]
@PyDataBlog null-ls is dead, i'd advise using conform.nvim for formatting https://github.com/stevearc/conform.nvim

**Reply by PyDataBlog** [reply]
thanks for the suggestion. how would I still add the ruff args to conform.nvim?

**Reply by siduck** [reply]
@PyDataBlog  i just use the default stuff in conform.nvim so idk, you can read its docs

**Reply by PyDataBlog** [reply]
Will do!
